src/utils/k6/client.ts

```diff
@@ -0,0 +1,197 @@
+import { ChildProcessWithoutNullStreams, spawn } from 'child_process'
+import { app } from 'electron'
+import path from 'path'
+import readline from 'readline/promises'
+import { z } from 'zod'
+
+import { LogEntrySchema } from '@/schemas/k6'
+import { K6Log } from '@/types'
+import { getArch, getPlatform } from '@/utils/electron'
+
+import { TestRun } from './testRun'
+
+const TestOptionsSchema = z.object({
+  cloud: z
+    .object({
+      name: z.string().nullish(),
+      projectID: z.number().nullish(),
+    })
+    .nullish(),
+})
+
+type TestOptions = z.infer<typeof TestOptionsSchema>
+
+const EXECUTABLE_NAME = getPlatform() === 'win' ? 'k6.exe' : 'k6'
+
+function getDefaultExecutablePath() {
+  // @ts-expect-error - fails because we're targeting node
+  const resourcesPath = import.meta.env.DEV
+    ? path.join(app.getAppPath(), 'resources', getPlatform())
+    : process.resourcesPath
+
+  return path.join(resourcesPath, getArch(), EXECUTABLE_NAME)
+}
+
+interface SpawnArgs {
+  args: Array<string[] | string | null | undefined | false>
+  env?: NodeJS.ProcessEnv
+}
+
+interface SpawnResult {
+  code: number | null
+  stdout: string[]
+  stderr: string[]
+}
+
+interface ArchiveArgs {
+  scriptPath: string
+  outputPath?: string
+}
+
+interface InspectArgs {
+  scriptPath: string
+}
+
+interface RunArgs {
+  path: string
+  quiet?: boolean
+  insecureSkipTLSVerify?: boolean
+  noUsageReport?: boolean
+  env?: Record<string, string>
+}
+
+export class ArchiveError extends Error {
+  code: number | null
+  stderr: K6Log[]
+
+  constructor(code: number | null, stderr: K6Log[]) {
+    super('Failed to archive script')
+
+    this.code = code
+    this.stderr = stderr
+  }
+}
+
+export class K6Client {
```

Comment:
I move the `K6Client` in a separate commit, but sadly GitHub (and I suppose Git) doesn't take that into account when doing diffs. The `archive` and `inspect` methods have only small modifications. The bulk of changes is the `run` method.